gapic-generator-java/src/main/java/com/google/api/generator/gapic/composer/comment/ServiceClientCommentComposer.java

```diff
@@ -215,6 +229,115 @@ public static List<CommentStatement> createRpcMethodHeaderComment(
     return comments;
   }
 
+  private static String createTableOfMethods(List<MethodAndVariants> methodAndVariantsList) {
+    String FLATTENED_METHODS =
+        "<p>\"Flattened\" method variants have the fields of the request type converted into function parameters to enable multiple ways to call the same method.</p>\n";
+    String REQUEST_OBJECT_METHODS =
+        "<p>Request object method variants only takes one parameter, a request object, which must be constructed before the call.</p>\n";
+    String CALLABLE_METHODS =
+        "<p>Callable method variants take no parameters and returns an immutable API callable object, which can be used to initiate calls to the service.</p>\n";
+    String ASYNC_METHODS =
+        "<p>Methods that return long-running operations have \"Async\" method variants that return `OperationFuture` which is used to track polling of the service.</p>\n";
+
+    StringBuilder tableBuilder = new StringBuilder();
+    tableBuilder
+        .append("<table>\n")
+        .append("   <tr>\n")
+        .append("     <th>Method</th>\n")
+        .append("     <th>Description</th>\n")
+        .append("     <th>Method Variants</th>\n");
+    for (MethodAndVariants method : methodAndVariantsList) {
+      tableBuilder
+          .append("   <tr>\n")
+          .append("     <td>")
+          .append(method.method)
+          .append("</td>\n")
+          .append("     <td>")
+          .append("<p>" + method.description + "</p>")
+          .append("</td>\n")
+          .append("     <td>\n");
+      if (method.hasRequestObjectVariants) {
+        tableBuilder
+            .append("     " + REQUEST_OBJECT_METHODS + "     ")
+            .append("<ul>\n")
+            .append("     <li>")
+            .append(String.join("\n     <li>", method.requestObjectVariants))
+            .append("\n")
+            .append("     </ul>")
+            .append("\n");
+      }
+      if (method.hasFlattenedVariants) {
+        tableBuilder
+            .append("     " + FLATTENED_METHODS + "     ")
+            .append("<ul>\n")
+            .append("     <li>")
+            .append(String.join("\n     <li>", method.flattenedVariants))
+            .append("\n")
+            .append("     </ul>")
+            .append("\n");
+      }
+      if (method.hasAsyncVariants) {
+        tableBuilder
+            .append("     " + ASYNC_METHODS + "     ")
+            .append("<ul>\n")
+            .append("     <li>")
+            .append(String.join("\n     <li>", method.asyncVariants))
+            .append("\n")
+            .append("     </ul>")
+            .append("\n");
+      }
```

Comment:
These lines see to do the same things so I wonder if it could make sense to just simplify it to a helper method that can be invoked a few times:

Method/ variables names just as an example:
```
private void generateUnorderedListMethodVariants(StringBuilder tableBuilder, String methodType, List<String> methodVariants) {
    if (!methodVariants.isEmpty()) {
        tableBuilder
            .append("     " + methodType + "     ")
            .append("<ul>\n")
            .append("     <li>")
            .append(String.join("\n     <li>", methodVariants))
            .append("\n")
            .append("     </ul>")
            .append("\n");
    }
}
```
or
```
private void generateUnorderedListMethodVariants(String methodType, List<String> methodVariants) {
    StringBuilder sb = new StringBuilder();
    if (!methodVariants.isEmpty()) {
        sb
            .append("     " + methodType + "     ")
            .append("<ul>\n")
            .append("     <li>")
            .append(String.join("\n     <li>", methodVariants))
            .append("\n")
            .append("     </ul>")
            .append("\n");
    }
    return sb.toString();
}
```

and just call `generateUnorderedListMethodVariants(tableBuilder, ..., ...)` for the different variants.

nit: Also, I wonder if there is a better way to format than manually adding in whitespace. Might be an easy way to format something slightly wrong.

Reply:
Thanks that makes sense! Done. 

As for the manual whitespace - I agree it's clunky; I'll experiment a bit to see if I can not include that whitespace at all, but Cloud RAD formatting is a bit finicky.